### ISSUE TITLE
MultipleGPULossCompute no longer works?

### ISSUE THREAD

**Issue by richielo** [harvardnlp/annotated-transformer#16]
Sorry for raising another issue again. I am trying to adapt this code to my own dataset. Everything works fine except for loss computation with multiple gpus, with an assertion error in parallel_apply. Any idea how to fix this? I am using pytorch 0.4.1. Thank you so much for your help 

` train_loss = run_epoch((rebatch(pad_idx, b) for b in train_iter), model_par, 
    MultiGPULossCompute(model.generator, criterion, devices=devices, opt=model_opt))
      File "main.py", line 75, in run_epoch
    loss = loss_compute(out, batch.trg_y, batch.ntokens)
  File "/home/richie/ErrorCorrection/optim.py", line 59, in __call__
    gen = nn.parallel.parallel_apply(generator, out_column)
  File "/home/richie/anaconda3/envs/python36/lib/python3.6/site-packages/torch/nn/parallel/parallel_apply.py", line 30, in parallel_apply
    assert len(modules) == len(inputs)
AssertionError
`

**Comment by benbijituo** [comment]
Have you solved this problem? @richielo I'm just the same
